components/layout_thread/lib.rs

```diff
@@ -1080,41 +1092,48 @@ impl LayoutThread {
 
         let element = match document.root_element() {
             None => {
+                let now = std_time::precise_time_ns();
+                if let ReflowGoal::LayoutQuery(_, time) = data.reflow_goal {
```

Comment:
We should move this out of this match, because it is not exclusive to the None case.